Dev

Hastur opened the pull request.

Commit:
Sprint 16. Refactoring: track search error wrapper, adapter binding, sharing extracted from settings.

Commit:
Sprint 16. Player screen MVVM and binding refactoring.

Commit:
Sprint 16. Search screen MVVM refactoring started.

Commit:
Sprint 16. Search screen MVVM and binding refactoring finished

Commit:
Sprint 16. Settings screen MVVM and binding refactoring.

Commit:
Sprint 16. Main and library screens binding refactoring.

anastasianadelshtehel commented on app/src/main/java/com/practicum/playlistmaker/search/track_search/ui/SearchActivity.kt:
```diff
@@ -8,144 +8,172 @@ import android.os.Looper
+                override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
+                    searchClear.isVisible = !searchInput.text.isNullOrEmpty()
                     emptyScreen.isVisible = false
+                    searchDebounce()
```
Логика дебаунса (searchDebounce, clickDebounce) и обработки кликов находится в Activity, можно перенести  в Domain

anastasianadelshtehel commented on app/src/main/java/com/practicum/playlistmaker/search/track_search/ui/SearchActivity.kt:
```diff
@@ -8,144 +8,172 @@ import android.os.Looper
+                if (clickDebounce()) startActivity(
+                    Intent(this@SearchActivity, PlayerActivity::class.java).putExtra(
+                        TRACK,
+                        Utils().serializeToJson(track)
```
Прямая работа с сериализация должна быть в Domain/Data слоях

anastasianadelshtehel commented on app/src/main/java/com/practicum/playlistmaker/player/ui/PlayerActivity.kt:
```diff
@@ -1,132 +1,129 @@
+            trackName.text = trackModel.trackName
+            artistName.text = trackModel.artistName
+            valueTrackTime.text =
+                SimpleDateFormat("mm:ss", Locale.getDefault()).format(trackModel.trackTimeMillis)
```
Форматирование времени SimpleDateFormat можно вынести в Domain

anastasianadelshtehel commented on app/src/main/java/com/practicum/playlistmaker/search/track_search/presentation/SearchViewModel.kt:
```diff
@@ -0,0 +1,70 @@
+
+    fun searchTrack(searchQuery: String) {
+        screenStateLiveData.value = SearchScreenState.Loading
+        searchInteractor.searchTrack(searchQuery, object : SearchInteractor.TrackConsumer {
```
Использование колбэка TrackConsumer лучше использовать LiveData

anastasianadelshtehel commented on app/src/main/java/com/practicum/playlistmaker/player/presentation/PlayerViewModel.kt:
```diff
@@ -0,0 +1,89 @@
+    fun getScreenStateLiveData(): LiveData<PlayerScreenState> = screenStateLiveData
+
+    private var playingStatusLiveData = MutableLiveData<PlayingStatus>()
+    fun getPlayingStatusLiveData(): LiveData<PlayingStatus> = playingStatusLiveData
```
Избыточные LiveData, можно объединить в одну State

anastasianadelshtehel commented on app/src/main/java/com/practicum/playlistmaker/search/track_search/ui/SearchActivity.kt:
```diff
@@ -8,144 +8,172 @@ import android.os.Looper
+                    binding.run {
+                        progressBar.isVisible = false
+                        trackList.isVisible = false
+                        emptyScreen.isVisible = true
```
повторяющийся код можно вынести в отдельную функцию

anastasianadelshtehel commented on app/src/main/java/com/practicum/playlistmaker/search/track_search/ui/SearchActivity.kt:
```diff
@@ -8,144 +8,172 @@ import android.os.Looper
+                    startActivity(
+                        Intent(this@SearchActivity, PlayerActivity::class.java).putExtra(
+                            TRACK,
+                            Utils().serializeToJson(track)
```
Прямое обращение к Utils() для сериализации, лучше перенести во ViewModel

Commit:
Sprint 16. Playing status live data merged to screen state live data.

Commit:
Sprint 16. Refactoring:
1. Playing status live data merged to screen state live data.
2. Data formatting moved from activity.
3. Serialization moved from activity.

Commit:
Sprint 16. Refactoring: consumer callback removed.

Commit:
Sprint 16. Refactoring: added new screen states for views visibility control.

Commit:
Sprint 16. Refactoring: debouncer moved from activity.
